openstudiocore/src/model/GeneratorPVWatts.cpp

```diff
@@ -0,0 +1,443 @@
+    OS_ASSERT(result);
+    return result;
+  }
+
+  void GeneratorPVWatts_Impl::resetModuleType() {
+    bool result = setString(OS_Generator_PVWattsFields::ModuleType, "");
+    OS_ASSERT(result);
+  }
+
+  bool GeneratorPVWatts_Impl::setArrayType(std::string arrayType) {
+    bool result = setString(OS_Generator_PVWattsFields::ArrayType, arrayType);
+    OS_ASSERT(result);
+    return result;
+  }
+
+  void GeneratorPVWatts_Impl::resetArrayType() {
+    bool result = setString(OS_Generator_PVWattsFields::ArrayType, "");
+    OS_ASSERT(result);
+  }
+
+  bool GeneratorPVWatts_Impl::setSystemLosses(double systemLosses) {
+    bool result = setDouble(OS_Generator_PVWattsFields::SystemLosses, systemLosses);
+    OS_ASSERT(result);
+    return result;
+  }
+
+  void GeneratorPVWatts_Impl::resetSystemLosses() {
+    bool result = setString(OS_Generator_PVWattsFields::SystemLosses, "");
+    OS_ASSERT(result);
+  }
+
+  bool GeneratorPVWatts_Impl::setTiltAngle(double tiltAngle) {
+    bool result = setDouble(OS_Generator_PVWattsFields::TiltAngle, tiltAngle);
+    OS_ASSERT(result);
+    return result;
+  }
+
+  void GeneratorPVWatts_Impl::resetTiltAngle() {
+    bool result = setString(OS_Generator_PVWattsFields::TiltAngle, "");
+    OS_ASSERT(result);
+  }
+
+  bool GeneratorPVWatts_Impl::setAzimuthAngle(double azimuthAngle) {
+    bool result = setDouble(OS_Generator_PVWattsFields::AzimuthAngle, azimuthAngle);
+    OS_ASSERT(result);
+    return result;
+  }
+
+  void GeneratorPVWatts_Impl::resetAzimuthAngle() {
+    bool result = setString(OS_Generator_PVWattsFields::AzimuthAngle, "");
+    OS_ASSERT(result);
+  }
+
+  bool GeneratorPVWatts_Impl::setSurface(const PlanarSurface& surface) {
+    bool result = false;
+    if (surface.optionalCast<Surface>()){
+      result = setPointer(OS_Generator_PVWattsFields::SurfaceName, surface.handle());
+    }
+    else if (surface.optionalCast<ShadingSurface>()){
+      result = setPointer(OS_Generator_PVWattsFields::SurfaceName, surface.handle());
+    }
+    OS_ASSERT(result);
+    return result;
+  }
+
+  void GeneratorPVWatts_Impl::resetSurface() {
+    bool result = setString(OS_Generator_PVWattsFields::SurfaceName, "");
+    OS_ASSERT(result);
+  }
+
+  bool GeneratorPVWatts_Impl::setGroundCoverageRatio(double groundCoverageRatio) {
+    bool result = setDouble(OS_Generator_PVWattsFields::GroundCoverageRatio, groundCoverageRatio);
+    OS_ASSERT(result);
+    return result;
+  }
+
+  void GeneratorPVWatts_Impl::resetGroundCoverageRatio() {
+    bool result = setString(OS_Generator_PVWattsFields::GroundCoverageRatio, "");
+    OS_ASSERT(result);
+  }
+
+} // detail
+
+GeneratorPVWatts::GeneratorPVWatts(const Model& model, double dcSystemCapacity)
+  : Generator(GeneratorPVWatts::iddObjectType(),model)
+{
+  OS_ASSERT(getImpl<detail::GeneratorPVWatts_Impl>());
+
+  bool ok = true;
+  OS_ASSERT(ok);
+
+  ok = setDCSystemCapacity(dcSystemCapacity);
```

Comment:
This should remove the object and throw if not ok (e.g. capacity < 0).  See this for an example:

https://github.com/NREL/OpenStudio/blob/develop/openstudiocore/src/model/PlanarSurface.cpp#L636